lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -865,3 +865,54 @@ def test_course_catalog_access_num_queries(self, user_attr_name, action, course_
         course_overview = CourseOverview.get_from_id(course.id)
         with self.assertNumQueries(num_queries, table_blacklist=QUERY_COUNT_TABLE_BLACKLIST):
             bool(access.has_access(user, action, course_overview, course_key=course.id))
+
+
+class CCXOverviewTestCase(SharedModuleStoreTestCase):
+    """
+    Tests for ccx course overviews
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def test_ccx_visibility(self):
+        """
+        Test to ensure that the ccx courses are hidden in course catalog.
+        """
+
+        course = CourseFactory.create(enable_ccx=True)
+        CourseOverview.load_from_module_store(course.id)
+
+        coach = AdminFactory.create()
+        ccx = CustomCourseForEdX(
+            course_id=course.id,
+            coach=coach,
+            display_name="Test CCX"
+        )
+        ccx.save()
+
+        student_password = "foobar"
+        student = UserFactory.create(username="test", password=student_password, is_staff=False)
+
+        ccx_overview = CourseOverview.get_from_id(ccx.locator)
```

Comment:
Interesting that you had to create a CourseOverview object here.. do the tests not work with just the `ccx` object you created above?  Ok if not, I'm just wondering.

Reply:
No - `has_access` explicitly doesn't handle the ccx objects directly - it requires the CourseOverview object or a CourseDescriptor for course based visibility checking.

Reply:
👍